source/__tests__/userController.test.ts

```diff
@@ -0,0 +1,143 @@
+import { Request, Response, NextFunction } from "express";
+import { mock, users } from "../models/__mocks__";
+import "regenerator-runtime/runtime";
+import constants from "../config/constants";
+
+import UserController from "../controllers/UserController";
+
+describe("UserController", () => {
+  let userService = {
+    getUsers: jest.fn(),
+    getUserById: jest.fn(),
+    createUser: jest.fn(),
+    updateUser: jest.fn(),
+    deleteUser: jest.fn(),
+    createDefaultUsers: jest.fn(),
+    getUserByLoginAndPassword: jest.fn(),
+  };
+
+  const request = mock.request();
+  const response = mock.response();
+  const next = mock.next();
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("Get All Users", async () => {
+
+    userService = {
+      ...userService,
+      getUsers: jest.fn().mockResolvedValue({ rows: users }),
+    };
+
+    await new UserController(userService).getUsers(
+      request as unknown as Request,
```

Comment:
you can modify it to       {} as Request,

Reply:
done